src/app/view/components/_common/dot-download-bundle-dialog/dot-download-bundle-dialog.component.spec.ts

```diff
@@ -241,17 +204,17 @@ describe('DotDownloadBundleDialogComponent', () => {
                     expect(cancelButton.disabled).toEqual(true);
                 });
 
-                it('should set location to the correct url when publish', () => {
+                it('should fetch to the correct url when publish', () => {
                     downloadButton.click();
-                    expect(document.location.href).toEqual(
+                    expect(window.fetch).toHaveBeenCalledWith(
                         `${DOWNLOAD_URL}${BUNDLE_ID}/operation/publish/filterKey/2`
                     );
                 });
                 it('should set location to the correct url when unplublish', () => {
                     unPublishButton.click();
                     fixture.detectChanges();
                     downloadButton.click();
-                    expect(window.location.href).toEqual(
+                    expect(window.fetch).toHaveBeenCalledWith(
```

Comment:
How can we test the download itself?, I mean we can remove the download part and no tests will break.

Reply:
after the last commit, this is tested in the previous tests.